### ISSUE TITLE
Graphical Selection: Add support for manual entry

### ISSUE THREAD

**Issue by XenonA7** [CCDirectLink/crosscode-map-editor#278]
The new graphical selection widget for NPCs, Enemies, props, etc is very cool and useful. However, it does not allow these values to be manually specified. Modders who have custom NPCs, enemies, etc will need to manually set these in the json file now.

Ideally the widget would allow manual string entry similar to how https://github.com/CCDirectLink/crosscode-map-editor/pull/233 allows manual string entry with dropdowns.

![image](https://github.com/CCDirectLink/crosscode-map-editor/assets/105614278/2c1e496a-0645-492c-a7b6-69d240b0a91c)

**Comment by XenonA7** [comment]
This is also required for DLC enemies, such as **beach.turtle-man**. Right now there is no longer a way to place one of these enemies without json editing.

**Comment by Vegita2** [comment]
did you try selecting the mod in the settings? It should also include all enemies/npcs/etc added by mods

**Comment by XenonA7** [comment]
Do we need to start selecting a mod from now on? I have never liked that feature because it prevents all vanilla maps from being in the quick list inside the editor.
(this list)
![image](https://github.com/CCDirectLink/crosscode-map-editor/assets/105614278/fa895d0f-5a58-43cd-8397-61f480810c5b)

I like having the vanilla maps always available to look at for references, and loading a mod removes this. I am also working on multiple mods that add custom enemies/npcs and having to switch between them is not ideal. Previous versions of the map editor allowed manual entry so this was not a problem.

**Comment by Vegita2** [comment]
> Do we need to start selecting a mod from now on? I have never liked that feature because it prevents all vanilla maps from being in the quick list inside the editor. I like having the vanilla maps always available to look at for references, and loading a mod removes this.

This can be a separate issue to always show the list 

> I am also working on multiple mods that add custom enemies/npcs and having to switch between them is not ideal

Selecting a mod should also load all of it's dependencies.
<br>
We can implement the basic string input to make it easier to work with currently, but for the long term we should try to get the mod support to a level so we don't need to rely on these basic inputs.

Can you describe what you would need from the mod support to actually prefer to use it? Based on your comment i got two additional features
1. show all maps instead of only maps from the mod
2. selecting multiple mods

**Comment by XenonA7** [comment]
I just wanted to say thank you for your work on this tool, it is invaluable for us modders. I think these two features are the main reasons why I prefer not to load mods in the map editor right now. Keep in mind that some mods, especially from newer modders, replace vanilla maps with their own versions of them, without patches. This is also the case when vanilla terrain needs to be modified because there is not currently an easy way to patch tile data (as far as I know). So I think it would be best to have separate dropdowns in the map list. All vanilla maps would be under one dropdown, there would also be one for each loaded mod containing all maps from each mod. Unless you think having all the vanilla and modded maps in the same list would be ideal.

But back to the original topic of this issue, the more pressing matter is the inability to place enemies from the post-game DLC.

**Comment by Vegita2** [comment]
there is a long standing issue to support extensions, #226

**Comment by XenonA7** [comment]
Yes, but this issue is about how after the update, it is no longer possible to place DLC or modded enemies/NPCs without doing it in json.

**Comment by Vegita2** [comment]
added new issues for the mod stuff, so this one can just focus on the basic string input
